app/models/short_url_request.rb

```diff
@@ -62,4 +69,8 @@ def strip_whitespace
     self.from_path = self.from_path.strip
     self.to_path = self.to_path.strip
   end
+
+  def should_republish_redirect?
+    accepted? && (from_path_changed? || to_path_changed?)
```

Comment:
Cool.  I think I'm remembering older versions of Rails where the dirty tracking didn't handle this use case. Good to have tests making this explicit though.